maven-artifact/src/main/java/org/apache/maven/artifact/ArtifactUtils.java

```diff
@@ -92,13 +92,22 @@ public static String key( Artifact artifact )
 
     public static String key( String groupId, String artifactId, String version )
     {
-        Validate.notBlank( groupId, "groupId can neither be null, empty nor blank" );
-        Validate.notBlank( artifactId, "artifactId can neither be null, empty nor blank" );
-        Validate.notBlank( version, "version can neither be null, empty nor blank" );
+        notBlank( groupId, "groupId can neither be null, empty nor blank" );
+        notBlank( artifactId, "artifactId can neither be null, empty nor blank" );
+        notBlank( version, "version can neither be null, empty nor blank" );
 
         return groupId + ":" + artifactId + ":" + version;
     }
 
+    private static void notBlank( String str, String message )
+    {
+        int c = str != null ? str.charAt( 0 ) : 0;
```

Comment:
Isn't this something which should be in Commons slang rather than here?

Reply:
Not sure, the optimization depends a bit on the context.  In the maven case, I've never seen a pom with whitespaces in the groupId / artifactId or such, so almost 100% of the tests returns false, which is why the optimization is easy.
In the general case, it may not be worth the change.
Btw, I need to fix that one as it should be
```
int c = ( str != null && str.length() > 0 ) ? str.charAt( 0 ) : 0;
```